packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -331,39 +333,68 @@ export function closeAllWorkspaces(onClosingWorkspaces: () => void = () => {}) {
   }
 }
 
-export function getWhetherWorkspaceCanBeClosed(name: string, ignoreChanges = false) {
-  const promptCheckFcn = getPromptBeforeClosingFcn(name);
-  return ignoreChanges || !promptCheckFcn || !promptCheckFcn();
-}
-
 type PromptType = 'closing-workspace' | 'closing-all-workspaces' | 'closing-workspace-launching-new-workspace';
 
-export function showWorkspacePrompts(promptType: PromptType, onConfirmation: () => void = () => {}) {
+/**
+ * Which type of prompt should be shown to the user.
+ * @param promptType 'closing-workspace' | 'closing-all-workspaces' | 'closing-workspace-launching-new-workspace'
+ * @param onConfirmation Function to be called after the user confirms to close the workspace
+ * @param workspaceTitle Workspace title to be shown in the prompt
+ * @returns
+ */
+
+export function showWorkspacePrompts(
```

Comment:
I don't really like this as a pattern; I think it's inappropriate unless the same prompts are being launched from many different places in the code. It is better to have the content of the prompt housed in the context that it is relevant to. But that's just sort of a note for posterity; let's get this PR in.